### PR TITLE
[master] 3.3.5 -> master cherry-pick

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Cherry picking

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

### CODE REVIEW

**Comment by msoky** [comment]
ready to merge?

**Comment by msoky** [comment]
and now we have conflicts...

**Comment by MaxtorCoder** [comment]
> and now we have conflicts...

Lol, calm 🤣

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -20656,7 +20656,7 @@ void Player::SaveToDB(LoginDatabaseTransaction loginTransaction, CharacterDataba
         stmt->setUInt32(index++, m_Played_time[PLAYED_TIME_TOTAL]);
         stmt->setUInt32(index++, m_Played_time[PLAYED_TIME_LEVEL]);
         stmt->setFloat(index++, finiteAlways(_restMgr->GetRestBonus(REST_TYPE_XP)));
-        stmt->setUInt64(index++, time(nullptr));
+        stmt->setUInt64(index++, uint64(GameTime::GetGameTime()));
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp:20659]: remove uint64 cast

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -20798,7 +20798,7 @@ void Player::SaveToDB(LoginDatabaseTransaction loginTransaction, CharacterDataba
...
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp:20801]: remove uint64 cast

File: src/server/game/Server/Packets/MailPackets.cpp [hunk 3]
```diff
@@ -15,6 +15,7 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "GameTime.h"
```
Comment (Shauren) [inline src/server/game/Server/Packets/MailPackets.cpp:18]: wrong include order

**Review by Shauren — APPROVED** [review]
ok merge

**Pull request merged**